node/flatpak-node-generator.py

```diff
@@ -619,11 +652,51 @@ def process_lockfile(self, lockfile: Path) -> Iterator[Package]:
         raise NotImplementedError()
 
 
+class RCFileProvider:
+    def get_node_headers(self, rcfile: Path) -> NodeHeaders:
+        raise NotImplementedError()
+
+
 class ModuleProvider(contextlib.AbstractContextManager):
     async def generate_package(self, package: Package) -> None:
         raise NotImplementedError()
 
 
+class NodeHeadersProvider(contextlib.AbstractContextManager):
+    def __init__(self, gen: ManifestGenerator):
+        self.gen = gen
+
+    def __exit__(self, *_: Any) -> None:
+        pass
+
+    def get_gyp_dir(self) -> Path:
+        return self.gen.data_root / 'cache' / 'node-gyp'
+
+    async def generate_node_headers(self, node_headers: NodeHeaders, dest: Path = None):
+        url = node_headers.get_url()
+        install_version = node_headers.get_install_version()
+        if dest is None:
+            dest = self.get_gyp_dir() / node_headers.target
+        metadata = await RemoteUrlMetadata.get(url, cachable=True)
+        self.gen.add_archive_source(url,
+                                    metadata.integrity,
+                                    destination=dest)
+        self.gen.add_data_source(install_version,
+                                 destination=dest / 'installVersion')
+
+    async def generate_sdk_node_headers(self):
+        gyp_dir = self.get_gyp_dir()
```

Comment:
no need to be async here, also a bit confused on the naming. So we have generate_node_headers, which generates the targets to download node headers, but also generate_sdk_node_headers which doesn't actually download new headers? From what I can tell, this is symlinking the global SDK extension node headers for Gyp's use (based on the argument string), so maybe something about symlink_global_sdk_headers or similar, just for a bit more clarity!